### PR TITLE
Series labels and diagram with definitions.

### PR DESCRIPTION
**Pull request by chrischute**

### CODE REVIEW

File: ARCharts/ARBarChart.swift [hunk 1]
```diff
@@ -99,42 +99,57 @@ public class ARBarChart: SCNNode {
...
+        guard spaceForSeriesLabels >= 0.0 && spaceForSeriesLabels <= 1.0 else {
+            fatalError("ARBarChartDelegate method spaceForSeriesLabels must return a value between 0.0 and 1.0")
+        }
+        let spaceForIndexLabels = delegate.spaceForIndexLabels(in: self)
+        guard spaceForIndexLabels >= 0.0 && spaceForIndexLabels <= 1.0 else {
+            fatalError("ARBarChartDelegate method spaceForIndexLabels must return a value between 0.0 and 1.0")
         }
         
+        
+
```
Comment (Boris-Em) [inline ARCharts/ARBarChart.swift]: Nit: Extra space

File: ARCharts/ARBarChart.swift [hunk 2]
```diff
@@ -198,6 +233,7 @@ public class ARBarChart: SCNNode {
      * - parameter series: The series that requires the Z position.
      * - parameter previousSeriesZPosition: The Z position of the previous series. This is useful for optimization.
      * - parameter seriesSize: The acutal size available for one series on the graph.
+     * - parameter zShift: The shift to add to each Z position to center the graph.
```
Comment (Boris-Em) [inline ARCharts/ARBarChart.swift]: Nit: This is not a parameter anymore

File: ARCharts/ARBarChart.swift [hunk 3]
```diff
@@ -143,6 +158,26 @@ public class ARBarChart: SCNNode {
             }
             
             previousZPosition = zPosition
+            
+            if let seriesLabelText = dataSource.barChart(self, labelForSeries: series) {
```
Comment (Boris-Em) [inline ARCharts/ARBarChart.swift:160]: Nit: Could you break that down into another function?
Something like `addLabels`?

File: ARCharts/ARBarChartProtocols.swift [hunk 4]
```diff
@@ -88,14 +88,32 @@ public protocol ARBarChartDelegate: class {
                   forSeries series: Int) -> UIColor
     
     /**
+     *  Asks the delegate to return the color for a label for a specific series (Z axis).
+     * - parameter barChart: The `ARBarChart` object requesting the number of values.
```
Comment (Boris-Em) [inline ARCharts/ARBarChartProtocols.swift]: Nit: Should be "requesting the color"

File: ARCharts/ARBarChartProtocols.swift [hunk 5]
```diff
@@ -88,14 +88,32 @@ public protocol ARBarChartDelegate: class {
...
+     * - parameter series: The index number identifying a series in the bar chart (Y axis).
+     * - returns: The color to use for the label corresponding to the given series.
+     */
+    func barChart(_ barChart: ARBarChart,
+                  colorForLabelForSeries series: Int) -> UIColor
+    
+    /**
+     *  Asks the delegate to return the color for a label for a specific index (X axis).
+     * - parameter barChart: The `ARBarChart` object requesting the number of values.
```
Comment (Boris-Em) [inline ARCharts/ARBarChartProtocols.swift]: Nit: Should be "requesting the color"

**Review by Boris-Em — APPROVED** [review]
Looks great! 
Feel free to merge after addressing the minor comments I left.

File: ARCharts/ARBarChart.swift (hunk 3 above)
Comment (chrischute) [reply ARCharts/ARBarChart.swift:160]: Yes, I'll do that when I add index labels.

**Pull request merged**